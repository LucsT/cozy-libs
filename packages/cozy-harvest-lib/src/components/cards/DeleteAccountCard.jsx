import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import Card from 'cozy-ui/transpiled/react/Card'
import { translate } from 'cozy-ui/transpiled/react/I18n'
import { SubTitle, Text } from 'cozy-ui/transpiled/react/Text'

import DeleteAccountButton from '../DeleteAccountButton'

export class DeleteAccountCard extends PureComponent {
  render() {
    const { account, onSuccess, t } = this.props
    return (
      <Card>
        <SubTitle className="u-mb-1">{t('card.deleteAccount.title')}</SubTitle>
        <Text className="u-mb-1">{t('card.deleteAccount.description')}</Text>
        <DeleteAccountButton
          account={account}
          onSuccess={onSuccess}
          extension="full"
        />
      </Card>
    )
  }
}

DeleteAccountCard.propTypes = {
  ...DeleteAccountButton.propTypes,
  t: PropTypes.func.isRequired
}

export default translate()(DeleteAccountCard)
