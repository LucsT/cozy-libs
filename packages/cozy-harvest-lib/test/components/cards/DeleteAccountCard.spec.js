import React from 'react'
import { shallow } from 'enzyme'

import DeleteAccountCard from 'components/cards/DeleteAccountCard'

const fixtures = {
  account: {
    _id: '1865a2e044414a15aebbad75ca96c5cc'
  }
}

const props = {
  account: fixtures.account,
  onSuccess: jest.fn(),
  t: key => key
}

describe('DeleteAccountCard', () => {
  it('should render', () => {
    const component = shallow(<DeleteAccountCard {...props} />).getElement()
    expect(component).toMatchSnapshot()
  })
})
