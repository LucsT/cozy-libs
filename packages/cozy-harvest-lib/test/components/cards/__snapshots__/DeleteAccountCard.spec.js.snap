// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAccountCard should render 1`] = `
<DeleteAccountCard
  account={
    Object {
      "_id": "1865a2e044414a15aebbad75ca96c5cc",
    }
  }
  onSuccess={[MockFunction]}
/>
`;
